### DISCUSSION TITLE
venv for js package

### DISCUSSION THREAD

**Discussion by anyoung-mc** [pyodide/pyodide#3189] [Q&A] [answered]
I can't seem to find the package that contains the `js` module used.  Is is it possible to pip install it for local development of scripts?

```py
from pyodide import create_proxy # Was able to pip install this
from js import document # Can't find this one?
```

**Comment by hoodmane** [comment, marked as answer]
Hi @anyoung-mc,

No, it's not possible to install the `js` module into a native Python. The `js` module is a mount point for the JavaScript global scope. So for instance, we don't define `document` inside of Pyodide. It just has the semantics of the JavaScript DOM `document` if it is present in the JavaScript global scope, which happens when you are running Pyodide in the main thread of a website. If you are running Pyodide in node or a Web Worker then `from js import document` will raise an import error because it is not present in these global scopes. In node you can do things like `from js.process import stdout` because the node global scope has a symbol called `process`.

It is possible to `pip install pyodide` and get `create_proxy` in a normal native Python. However, it doesn't really work. The reason we have set it up to allow this is:
1. mypy
2. documentation generation
3. native tests for functionality that does not use `create_proxy` but is in a file that imports `create_proxy` at top level
